tiny_dnn/core/framework/tensor.h

```diff
@@ -61,45 +63,173 @@
 
 namespace tiny_dnn {
 
+template<typename U = float_t>
+class TensorStorage {
+ public:
+    /**
+     * Initializes an empty tensor storage.
+     * @return
+     */
+    TensorStorage() {
+    }
+
+    /**
+     * Constructor that assepts a vector of shape and create a TensorStorage
+     * with a size equivalent to that shape.
+     * @param shape array containing N integers, sizes of dimensions
+     * @return
+     */
+    explicit TensorStorage(const std::vector<size_t> &shape) {
+        resize(shape);
+    }
+
+    /**
+     * Constructor that assepts an initializer list  of shape and create a
+     * TensorStorage with a size equivalent to that shape.
+     * @param shape array containing N integers, sizes of dimensions
+     * @return
+     */
+    explicit TensorStorage(std::initializer_list<size_t> const &shape) {
+        std::vector<size_t> tmp(shape.size());
+        std::copy(shape.begin(), shape.end(), tmp.begin());
+        resize(tmp);
+    }
+
+    void sync() {
+        if (data_is_on_host_)
+            toDevice();
+        else
+            fromDevice();
+    }
+
+    typename std::vector<U, aligned_allocator < U, 64>>::
+    iterator host_data(size_t offset) {
```

Comment:
The iterator should be typedef-ed so that you don't need to use `std::vector<U, aligned_allocator < U, 64>>` again and again.